tests/test_model.py

```diff
@@ -217,6 +217,21 @@ def test_product_nodata_nan() -> None:
     assert product.measurements["_neg_inf"].nodata == -numpy.inf
 
 
+def test_product_nodata_validation() -> None:
+    product_def = mk_sample_product(
+        "test",
+        measurements=[
+            {"name": "_nan", "dtype": "uint8", "nodata": "NaN"},
```

Comment:
I think it's worth making a second product that has `-100` as `nodata` with the `uint8` dtype and making sure the validation fails for that as well.